sdk/analytics/src/test/java/com/klaviyo/analytics/state/KlaviyoStateTest.kt

```diff
@@ -54,6 +54,26 @@ internal class KlaviyoStateTest : BaseTest() {
         job2.start()
     }
 
+    @Test
+    fun `Observer can detach itself during callback`() = runTest {
+        var observer: StateObserver = { _, _ -> }
+        var didRun = false
+
+        observer = { _, _ ->
+            state.offStateChange(observer)
+            didRun = true
+        }
+
+        state.onStateChange(observer)
+
+        state.reset()
+        assert(didRun) { "Observer did not run as expected" }
+
+        didRun = false
+        state.reset()
+        assert(!didRun) { "Observer should not run the second time" }
+    }
+
```

Comment:
Created a failing test for the issue I observed in the in-app forms company ID observer: calling `offStateChange` from within the observer would thrown an exception despite our earlier concurrency fixes.